langkit/lkt_lowering.py

```diff
@@ -1630,6 +1659,37 @@ def create_enum_node_alternatives(
         enum_node._alternatives_map = {alt.base_name.camel: alt.alt_node
                                        for alt in alternatives}
 
+    def create_enum(self,
+                    decl: L.EnumTypeDecl,
+                    annotations: EnumAnnotations) -> EnumType:
+        """
+        Create an EnumType instance.
+
+        :param decl: Corresponding declaration node.
+        :param annotations: Annotations for this declaration.
+        """
+        # Decode the list of enum literals and validate them
+        value_names = []
+        for lit in decl.f_literals:
+            name = lit.f_syn_name.text
+            check_source_language(
+                name not in value_names,
+                'The "{}" literal is present twice'
+            )
+            value_names.append(name)
+
+        check_source_language(
```

Comment:
Seems to me you'd go faster checking this as a whitelist: Only nodes can implement traits for the moment, and only built-in traits.

Reply:
I don’t understand your suggestion: how could we use a whitelist while no trait is allowed? The whitelist would be empty, right? In any case, I just realized that TokenNode should be legal for enum types, so I guess I should just remove this check.

Reply:
Never mind, this is an enum type, not an enum node: no trait should be allowed.

Reply:
As discussed live: let’s factorize the check for the presence/absence of traits to `resolve_type_decl` to avoid repeating it in all the `create_*` helpers.